### PR TITLE
fix(ivy): ensure component/directive `class` selectors are properly understood

### PR DESCRIPTION
**Pull request by matsko**
Angular allows for `<ng-content>` elements to include a selector which
filters which content-projected entries are inserted into the container
depending on whether or not the selector is matched.

With Ivy this feature has not fully worked due to the massive changes
that took place inside of Ivy's styling algorithm code (which is
responsible for assigning classes and styles to an element). This
fix ensures that content-projection can correctly identify which slot
an element should be placed into when class-based selectors are used.

### CODE REVIEW

**Review by mhevery — APPROVED** [review]

**Review by kara — CHANGES_REQUESTED** [review]
Can you remove the fixmes for FW-833 (or replace desc with new root cause if there are additional issues)?

**Review by kara — APPROVED** [review]
~~LGTM~~

**Review by kara — CHANGES_REQUESTED** [review]
@matsko Actually it looks like the tests added (for FW-833) are failing on CI. Can you take a look?

File: packages/core/test/linker/projection_integration_spec.ts [hunk 1]
```diff
@@ -81,22 +81,35 @@ describe('projection', () => {
...
-          set: {
-            template: '<multiple-content-tags>' +
-                '<div>B</div>' +
-                '<div>C</div>' +
-                '<div class="left">A</div>' +
-                '</multiple-content-tags>'
-          }
-        });
-        const main = TestBed.createComponent(MainComp);
+  fit('should project a single class-based tag', () => {
```
Comment (pkozlowski-opensource) [inline packages/core/test/linker/projection_integration_spec.ts]: `fit` left

**Review by kara — APPROVED** [review]
LGTM

**Comment by matsko** [comment]
https://tap.corp.google.com/ui#id=OCL:228943859:BASE:228949472:1547246925594:82175f07

**Comment by matsko** [comment]
http://test/OCL:228943859:BASE:229271551:1547509864607:f18a0f5

**Pull request closed without merging**